### ISSUE TITLE
Copy Form - Issues needed for Alfresco

### ISSUE THREAD

**Issue by gsteimer** [tsgrp/HPI#1962]
The copy form action was created for a Documentum client.  Items listed here are needed to get this functionality working on Alfresco.

- [x] Somehow, after copying a form, the new form is listed as a workflow document to the original form (see below).
- [x] When copying a form, the interface refreshes to the new form, rather than sticking to the original form.  The user should be able to choose whether to stay on the old form (close modal) or navigate to the new form (click the link in the confirmation message).

![image](https://user-images.githubusercontent.com/1377228/66312133-aa0a7180-e8d5-11e9-95c4-c90086721a39.png)

**Comment by ebrent** [comment]
COPY WIZARD FORM - fixed outstanding items such as: not refreshing the new form once copied in the background, , showing form's name as the clickable link instead of the form type, modal appearance, and workflow summary page to show that the form is incomplete if it cleared out supporting docs or user input questions. Biggest change was to change workflow relation to configurable parent/child relation via admin (dependent on alf or dctm). This required adding a custom config template.  Updated test files to pass accordingly.

CR: nwoodbine, aking

OC revision #23430
OCMS revision #23846

**Comment by gsteimer** [comment]
Copying a form now works on alfresco, however if you try to edit and finish the new form, it throws the following errors:

```
2019-10-10 13:29:37,551	ajp-apr-8009-exec-7	ERROR	RESTService	Unhandled exception around OC method class: interface com.tsgrp.opencontent.core.content.IContent| name: enhancedUpdateObject
com.tsgrp.opencontent.core.exception.OCRuntimeException: Unhandled exception around OC method class: interface com.tsgrp.opencontent.core.content.IContent| name: enhancedUpdateObject
	at com.tsgrp.opencontent.alfresco.core.EmbeddedAlfrescoAdvice.invoke(EmbeddedAlfrescoAdvice.java:98) ~[oc-hpiDemoAlf.2.alfrescoEmb-3.2.0.jar:3.2.0]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172) ~[spring-aop-3.2.14.RELEASE.jar:3.2.14.RELEASE]
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:204) ~[spring-aop-3.2.14.RELEASE.jar:3.2.14.RELEASE]
	at com.sun.proxy.$Proxy462.enhancedUpdateObject(Unknown Source) ~[?:?]
	at com.tsgrp.opencontent.wizard.core.Wizard2Impl.savePageSetInstance(Wizard2Impl.java:678) ~[oc-hpiDemoAlf.3.wizard-3.2.0.jar:3.2.0]
	at com.tsgrp.opencontent.wizard.action.executer.FinishAndSubmitPSIActionExecutor.executeImpl(FinishAndSubmitPSIActionExecutor.java:99) ~[oc-hpiDemoAlf.3.wizard-3.2.0.jar:3.2.0]
	at com.tsgrp.opencontent.core.action.executer.AbstractActionExecuter.execute(AbstractActionExecuter.java:224) ~[oc-hpiDemoAlf.1.core-3.2.0.jar:3.2.0]
	at com.tsgrp.opencontent.core.action.executer.AbstractActionExecuter.evaluateAndExecute(AbstractActionExecuter.java:154) ~[oc-hpiDemoAlf.1.core-3.2.0.jar:3.2.0]
	at com.tsgrp.opencontent.core.action.service.CoreActionService.execute(CoreActionService.java:475) ~[oc-hpiDemoAlf.1.core-3.2.0.jar:3.2.0]
	at com.tsgrp.opencontent.core.action.service.RESTAction.executeActionPost(RESTAction.java:224) ~[oc-hpiDemoAlf.1.core-3.2.0.jar:3.2.0]
...
Caused by: org.springframework.dao.ConcurrencyFailureException: Unexpected: current version does not appear to be 1st version in the list  [workspace://version2Store/10b971e0-3739-4035-be89-8fc57fc49165, workspace://SpacesStore/ab59e180-250e-427b-b478-ae0e15aa3142]
...
	at com.tsgrp.alfresco.content.util.dao.AlfrescoDaoUtils$1.doWork(AlfrescoDaoUtils.java:63) ~[oc-hpiDemoAlf.2.alfrescoEmb-3.2.0.jar:3.2.0]
	at com.tsgrp.alfresco.content.util.dao.AlfrescoDaoUtils$1.doWork(AlfrescoDaoUtils.java:59) ~[oc-hpiDemoAlf.2.alfrescoEmb-3.2.0.jar:3.2.0]
	at org.alfresco.repo.security.authentication.AuthenticationUtil.runAs(AuthenticationUtil.java:555) ~[alfresco-data-model-5.1.2.jar:5.1.2]
	at com.tsgrp.alfresco.content.util.dao.AlfrescoDaoUtils.synchronizeVersionStoreNodeProperties(AlfrescoDaoUtils.java:59) ~[oc-hpiDemoAlf.2.alfrescoEmb-3.2.0.jar:3.2.0]
	at com.tsgrp.opencontent.alfresco.util.AlfrescoEmbUtil.updateVersionHistory(AlfrescoEmbUtil.java:510) ~[oc-hpiDemoAlf.2.alfrescoEmb-3.2.0.jar:3.2.0]
	at com.tsgrp.opencontent.alfresco.event.events.AlfrescoVersionHistoryUpdateEvent.onUpdateObject(AlfrescoVersionHistoryUpdateEvent.java:35) ~[oc-hpiDemoAlf.2.alfrescoEmb-3.2.0.jar:3.2.0]
	at com.tsgrp.opencontent.core.event.OCPolicyComponentImpl.executeAfterUpdateObjectEvents(OCPolicyComponentImpl.java:174) ~[oc-hpiDemoAlf.1.core-3.2.0.jar:3.2.0]
	at com.tsgrp.opencontent.core.content.AbstractContentImpl.enhancedUpdateObject(AbstractContentImpl.java:220) ~[oc-hpiDemoAlf.1.core-3.2.0.jar:3.2.0]

```

**Comment by gsteimer** [comment]
This issue was resolved with the fix for #1969

**Comment by ebrent** [comment]
ACTIVE FORM - in revision #23846, I made some changes to psiview and activeform that aided the validation process on "edit form" of a copy form... However, that affected creating a new form negatively, by validating before even filling out a form (therefore showing error messages). It was decided that for now, these changes will be reverted. The outstanding issue is now, once you edit a copied form, there are green checkmarks even though the questions might not all be answered. However, in the summary page, it won't let you complete that form if there are unanswered questions (which we want). if you click back to a question from the summary page, they will then correctly validate an unanswered question. 

CR: skovancinski, nreiman
REVISION #23947
